packages/block-library/src/file/index.php

```diff
@@ -16,33 +16,39 @@
  * @return string Returns the block content.
  */
 function render_block_core_file( $attributes, $content ) {
+	if ( empty( $attributes['displayPreview'] ) ) {
+		return $content;
+	}
+
 	// If it's interactive, enqueue the script module and add the directives.
-	if ( ! empty( $attributes['displayPreview'] ) ) {
-		wp_enqueue_script_module( '@wordpress/block-library/file/view' );
+	wp_enqueue_script_module( '@wordpress/block-library/file/view' );
 
-		$processor = new WP_HTML_Tag_Processor( $content );
-		$processor->next_tag();
+	$processor = new WP_HTML_Tag_Processor( $content );
+	if ( $processor->next_tag() ) {
 		$processor->set_attribute( 'data-wp-interactive', 'core/file' );
-		$processor->next_tag( 'object' );
-		$processor->set_attribute( 'data-wp-bind--hidden', '!state.hasPdfPreview' );
-		$processor->set_attribute( 'hidden', true );
-
-		$filename     = $processor->get_attribute( 'aria-label' );
-		$has_filename = ! empty( $filename ) && 'PDF embed' !== $filename;
-		$label        = $has_filename ? sprintf(
-			/* translators: %s: filename. */
-			__( 'Embed of %s.' ),
-			$filename
-		) : __( 'PDF embed' );
-
-		// Update object's aria-label attribute if present in block HTML.
-		// Match an aria-label attribute from an object tag.
-		$processor->set_attribute( 'aria-label', $label );
-
-		return $processor->get_updated_html();
 	}
 
-	return $content;
+	// If there are no OBJECT elements, something might have already modified the block.
+	if ( ! $processor->next_tag( 'OBJECT' ) ) {
+		return $content;
```

Comment:
something I didn’t notice the first time around is that if we don’t find an OBJECT element, we also don’t set the `data-wp-interactive` attribute. this may be what we intend, but I wanted to point it out because I see it now

Reply:
performance shouldn’t be in question here, only behavior. so for example, if you wanted _every_ file block to get the Interactivity attribute, then that wouldn’t be the case if they contain no `OBJECT`.

I _think_ that’s intended, so it should be fine, but I wanted to note that as a thing to double-check.

the point is that even if we call `set_attribute()` we won’t see that applied unless we use or return `$processor->get_updated_html()`. it is practically “free” to set an attribute and then abandon the processor though.

Reply:
As far as know it should be only interactive if the File block displays PDF preview, which won’t be the case when object element isn’t present.

If a plugin fully replaces the output then they should handle it, or try and match the core block structure.

I don’t have a strong preference here, if you have a suggestion to improve and showcase the best practices, I would gladly apply it 🙇

Reply:
Looks all good from my vantage point 👍